soc/arm/atmel_sam0/samd20/CMakeLists.txt

```diff
@@ -3,4 +3,4 @@
 # Copyright (c) 2018 Sean Nyekjaer
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_sources(soc.c)
+zephyr_sources(../common/soc_samd2x.c)
```

Comment:
I've changed it so that `soc_samd2x.c` now gets selected based on whether ´CONFIG_SOC_SERIES_SAMD20`, `CONFIG_SOC_SERIES_SAMD21` or `CONFIG_SOC_SERIES_SAMR21` is set.

samd20/, samd21/ and samr21/ only contain header files and KConfig now.